### ISSUE TITLE
could you add a cache function for base64 skulls?

### ISSUE THREAD

**Issue by IceBear003** [Majekdor/SkullCache#3]
thank you for your nice plugin. It works very well in my server,
But I want to cache skulls which are generated from base64.
Could you update your plugin to support for that?

Thank you again, sorry for my bad English :D

**Comment by Majekdor** [comment]
So you're creating a skull from base 64 and you want to be able to cache it with this api?

**Comment by IceBear003** [comment]
> So you're creating a skull from base 64 and you want to be able to cache it with this api?

yes, guis in my server is full of base64 skulls. when a player join in and open the menu for the first time. The BossShopPro get 
 piles of skulls from texture.minecraft.com. And it cause an annoying lag in the main thread.

**Comment by IceBear003** [comment]
![image](https://user-images.githubusercontent.com/46891330/149899516-9d998169-ab42-433a-94e0-8c6cfcd83ad1.png)
this is one of the menus in my server.
I want to use your plugin to cache these skulls so that the server won't ask for skulls every time when a player opens it.

**Comment by Majekdor** [comment]
Ah. Well this isn't a plugin. It's an api that would allow the developer of BossShopPro to cache the fetched skulls.

**Comment by IceBear003** [comment]
> Ah. Well this isn't a plugin. It's an api that would allow the developer of BossShopPro to cache the fetched skulls.

I know, I want to write a plugin that depends on your API to cache these skulls when the server has started.

**Comment by Majekdor** [comment]
Ah okay. Do you need a way to cache item stacks or do you need to pass thru the base 64, have the api fetch the texture, and then cache it?

**Comment by IceBear003** [comment]
> you need to pass thru the base 64, have the api fetch the texture
pass thru the base 64, have the api fetch the texture, and then cache it
Just dont request these skulls again and again from texture.minecraft.net
I want to finish this at first.

**Comment by IceBear003** [comment]
> Ah okay. Do you need a way to cache item stacks or do you need to pass thru the base 64, have the api fetch the texture, and then cache it?

If you could update this plugin for me
I feel really grateful and dont know what to do to express my thanks

**Comment by Majekdor** [comment]
What are the skulls keyed to? Like how do you plan to pull them from the cache?

**Comment by IceBear003** [comment]
use base64 to pull them from the cache
and i dont know how the server send those skulls to clients: by sending textures or base64s or playerprofiles?
 if it works by sending textures, i want to cache the .png images and dont ask them from texture.minecraft.net again
if it works by sending base64s or playerprofiles, i want to send them directly to clients instead of requesting mojangapi
